labgrid/resource/udev.py

```diff
@@ -289,29 +289,48 @@ class USBSDMuxDevice(USBResource):
     """The USBSDMuxDevice describes an attached USBSDMux device,
     it is identified via USB using udev
     """
+
+    # Explanation: The USBSDMux consists of two devices, one block and one scsi
+    # generic device. The manager will mark the resource available as soon as
+    # the parent device is added. But the SDMux has not been fully initialized
+    # yet. Thats why this Device overwrites the avail attribute with an
+    # internal avail.
+    # The internal avail has to be an attribute, since attrs can only resolve
+    # its attributes during validation.
+    internal_avail = attr.ib(default=False)
+
     def __attrs_post_init__(self):
         self.match['ID_VENDOR_ID'] = '0424'
         self.match['ID_MODEL_ID'] = '4041'
         self.control_path = None
         self.disk_path = None
         super().__attrs_post_init__()
 
-    def update(self):
-        block_found = False
+    # Overwrite the avail attribute with our internal property
+    @property
+    def avail(self):
+        return self.internal_avail
```

Comment:
Can't we just check self.disk_path and self.control_path here instead of adding another attr above?